config/routes.rb

```diff
@@ -23,4 +23,7 @@
       delete :unlike
     end
   end
+
+  resources :reports
+  resources :punishments
```

Comment:
you don't have show and destroy routes. So add an except clause and remove them from routes as well.